core/resources.rs

```diff
@@ -81,7 +85,12 @@ impl ResourceTable {
 /// Abstract type representing resource in Deno.
 pub trait Resource: Downcast + Any + Send {
   /// Method that allows to cleanup resource.
-  fn close(&self) {}
+  ///
+  /// Resource can return vector of subresource ids that should
+  /// be closed as well.
```

Comment:
Hrm... I don't follow what's happening in this patch. What's a subresource?

Reply:
I didn't want to copy-paste a lot of "accept" logic for TLS (because it also uses TcpStream) so I made TlsListenerResource to hold reference to TcpListenerResource. 

I'm not adamant about it, but I didn't find any other way not to duplicate Accept future as AcceptTLS

Reply:
I'm not so happy with having this Resource::close method at all.  I think this is important to get right. I added some comments about the direction I'm thinking of in https://github.com/denoland/deno/pull/3274 WDYT?

Reply:
Yeah sure, using `drop` instead of `close` makes sense. Only thing is I won't be able to reference `TcpListener` from `TlsListener` but that's not a big deal.